backend/src/apiserver/server/experiment_server_test.go

```diff
@@ -1,116 +1,133 @@
-	_, err := server.CreateExperimentV1(nil, &apiV1beta1.CreateExperimentRequest{Experiment: experiment})
+	_, err := server.CreateExperimentV1(nil, &apiv1beta1.CreateExperimentRequest{Experiment: experiment})
 	assert.NotNil(t, err)
-	assert.Contains(t, err.Error(), "Create experiment failed.")
+	assert.Contains(t, err.Error(), "Failed to add experiment to experiment table")
 }
 
 func TestCreateExperiment_Failed(t *testing.T) {
 	clientManager := resource.NewFakeClientManagerOrFatal(util.NewFakeTimeForEpoch())
-	resourceManager := resource.NewResourceManager(clientManager)
+	resourceManager := resource.NewResourceManager(clientManager, map[string]interface{}{"DefaultNamespace": "default", "ApiVersion": "v2beta1"})
 	server := ExperimentServer{resourceManager: resourceManager, options: &ExperimentServerOptions{CollectMetrics: false}}
 	experiment := &apiV2beta1.Experiment{DisplayName: "ex1", Description: "first experiment"}
 	clientManager.DB().Close()
 	_, err := server.CreateExperiment(nil, &apiV2beta1.CreateExperimentRequest{Experiment: experiment})
 	assert.NotNil(t, err)
-	assert.Contains(t, err.Error(), "Create experiment failed.")
+	assert.Contains(t, err.Error(), "Failed to add experiment to experiment table")
 }
 
-func TestCreateExperimentV1_SingleUser_NamespaceNotAllowed(t *testing.T) {
-	clientManager := resource.NewFakeClientManagerOrFatal(util.NewFakeTimeForEpoch())
-	resourceManager := resource.NewResourceManager(clientManager)
-	server := ExperimentServer{resourceManager: resourceManager, options: &ExperimentServerOptions{CollectMetrics: false}}
-	resourceReferences := []*apiV1beta1.ResourceReference{
-		{
-			Key:          &apiV1beta1.ResourceKey{Type: apiV1beta1.ResourceType_NAMESPACE, Id: "ns1"},
-			Relationship: apiV1beta1.Relationship_OWNER,
-		},
-	}
-	experiment := &apiV1beta1.Experiment{
-		Name:               "exp1",
-		Description:        "first experiment",
-		ResourceReferences: resourceReferences,
-	}
-
-	_, err := server.CreateExperimentV1(nil, &apiV1beta1.CreateExperimentRequest{Experiment: experiment})
-	assert.NotNil(t, err)
-	assert.Contains(t, err.Error(), "In single-user mode, CreateExperimentRequest shouldn't contain resource references.")
-}
-
-func TestCreateExperiment_SingleUser_NamespaceNotAllowed(t *testing.T) {
-	clientManager := resource.NewFakeClientManagerOrFatal(util.NewFakeTimeForEpoch())
-	resourceManager := resource.NewResourceManager(clientManager)
-	server := ExperimentServer{resourceManager: resourceManager, options: &ExperimentServerOptions{CollectMetrics: false}}
-	experiment := &apiV2beta1.Experiment{DisplayName: "exp1", Description: "first experiment", Namespace: "ns1"}
-	_, err := server.CreateExperiment(nil, &apiV2beta1.CreateExperimentRequest{Experiment: experiment})
-	assert.NotNil(t, err)
-	assert.Contains(t, err.Error(), "In single-user mode, CreateExperimentRequest shouldn't contain namespace.")
-}
+// func TestCreateExperimentV1_SingleUser_NamespaceNotAllowed(t *testing.T) {
```

Comment:
Uncomment or remove the dead code.

Reply:
This is about creating source, are we no longer rejecting namespace input in single user mode? I think we probably should. Even if we don't filter on retrieving, we shouldn't allow storing arbitrary value that may or may not be valid.